### ISSUE TITLE
We need to ask for consent before enabling our SDK

### ISSUE THREAD

**Issue by HazAT** [getsentry/sentry-mobile#150]
We need some kind of popup that asks for consent to enable our SDK. Even for crash reporting, we need that.

Here is what Google says:

```
In cases where users may not reasonably expect that their personal or sensitive user data will be required to provide or improve the policy compliant features or functionality within your app (e.g., data collection occurs in the background of your app), you must meet the following requirements:
You must provide an in-app disclosure of your data access, collection, use, and sharing. The in-app disclosure:
Must be within the app itself, not only in the app description or on a website;
Must be displayed in the normal usage of the app and not require the user to navigate into a menu or settings;
Must describe the data being accessed or collected;
Must explain how the data will be used and/or shared;
Cannot only be placed in a privacy policy or terms of service; and
Cannot be included with other disclosures unrelated to personal or sensitive data collection.
Your in-app disclosure must accompany and immediately precede a request for user consent and, where available, an associated runtime permission. You may not access or collect any personal or sensitive data until the user consents. The app's request for consent:
Must present the consent dialog clearly and unambiguously;
Must require affirmative user action (e.g. tap to accept, tick a check-box);
Must not interpret navigation away from the disclosure (including tapping away or pressing the back or home button) as consent; and
Must not use auto-dismissing or expiring messages as a means of obtaining user consent.
```

**Comment by HazAT** [comment]
So we need to ask for it somewhere and store the decision. Also, make it opt-in in the future (in settings).

During onboarding, remove the SKIP button.

After the "Before we start..." screen, add another screen that shows an image and asks for consent.

"One last thing"

We are using our own SDK for collecting crash reports and session data and to improve our product. Are we allowed to? Pleaeassseee

(Two buttons)

Disable SDK (Grey)     Enable SDK (Purple)

**Comment by HazAT** [comment]
In settings, it should only be a radio button switch or checkbox.
If the user activates while not being activated, we should call `Sentry.init`

If the user deactivates this while it was activated, we should have a gatekeeper in `before_send` that checks for this flag and discards events from being sent.

During rehydration, we check if the stored decision and `init` (or not) the SDK.

**Comment by denrase** [comment]
This was closed by #153
